Assets/Scripts/AssetPlacement.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Asset_placement : MonoBehaviour
+{
+    public Mesh hexagonMesh;
+    public List<GameObject> floorChildren;
+    private string floorModelDirectory = "";
+    #region test_variables
+    public float testUpperHeightLimit = 0.4f;
+    public float testLowerHeightLimit = 0f;
+    public int testCollumns = 10;
+    public int testRows = 10;
+    #endregion
+
+    // Variables for distance calculation
+    private Bounds hexagonBounds;
+    private float x;
+    private float y;
+    private int scale = 100; // scale of hexagon
+
+    void Start()
+    {
+        hexagonMesh = GetComponent<MeshFilter>().mesh;
+        hexagonBounds = hexagonMesh.bounds;
+        x = (hexagonBounds.max.x * 2) * scale;     // Coordinate offset
+        y = (hexagonBounds.max.y * 3 / 2) * scale; // calculations
+
+        floorChildren = new List<GameObject>();
+        placeTest(testCollumns, testRows);
+    }
+    void placeTest(int collumns, int rows)
+    {
+        for (int i = 0; i < collumns; i++)
+        {
+            for (int j = 0; j < rows; j++)
+            {
+                if (i % 2 != 0)
+                {
+                    placeTile(new Vector3(j * x + x / 2, UnityEngine.Random.Range(testUpperHeightLimit, testLowerHeightLimit), i * y), "defaultHex");
+                    Debug.Log(String.Format(j * x + x / 2 + " " + i * y));
+                }
+                else
+                    placeTile(new Vector3(j * x, UnityEngine.Random.Range(testUpperHeightLimit, testLowerHeightLimit), i * y), "defaultHex");
+            }
+        }
+    }
+    void placeTile(Vector3 position, string hexBlockName)
+    {
+        hexBlockName = floorModelDirectory + hexBlockName;
+        GameObject hex = Resources.Load<GameObject>(hexBlockName);
+        if (hex != null)
+        {
+            floorChildren.Add(Instantiate<GameObject>(hex, position, Quaternion.LookRotation(Vector3.up, Vector3.forward), transform));
```

Comment:
Reiktų nuspręst ar mes norim spawninti objektus su Instantiate. Jei pastoviai tai darom, gali kainuoti performance.
https://discussions.unity.com/t/is-instantiating-bullets-many-objects-always-bad-for-performance/70097
Teoriškai jei tai darom tik pradžioje, tai kaip ir ok, bet klasė kiek suprantu bendra (AssetPlacement), tai su kokiais bulletais ar panašiai gali būt negerai.

Reply:
Tai manau ten nebus tiek daug tų tiles, jeigu ką tai disablinsim objektus kai išeis iš fov. Manrods tas object pooling labiau skirtas vienodiem objektam, o mes gal skirtingas tiles darysim (pvz  water, sand, grass)?

Reply:
Su tiles manau kaip ir ok, bet klausimas dėl kitokių objektų kaip armijos, bulletai, particles ir panašiai, kuriuos spawninsim vėliau su šita klase.